src/models/extensions/defaultExtension.ts

```diff
@@ -0,0 +1,8 @@
+import { FileFormat } from './fileFormat';
+
+export interface IDefaultExtension {
+  icon: string; // name of the icon.
+  format: FileFormat | string; // format of the icon
+  _custom?: boolean; // user customization: flag for the icon generator.
```

Comment:
I thought about this too. I don't usually like to use this kind of syntax but I wanted to make evident that this field was not meant to be edited by users. As I don't know users I thought that this was the easiest path for making them know that this property is not intended for modification. I don't rely on them writing the docs so I thought that this kind of syntax would put them in the right track.